### ISSUE TITLE
Investigate performance regression

### ISSUE THREAD

**Issue by cmyr** [googlefonts/fontc#369]
> is almost 4 times slower than when I also revert https://github.com/googlefonts/fontc/pull/359

maybe this warrants its own issue. Something is going on with #359 that has slowed down things noticeably. Building GoogleSans.designspace takes 27 seconds after #359, and only 5 seconds if I revert it (timings are with fontc built in release mode and with --emit-ir=false, debug or emit-ir=true make it even slower). I am not able to figure out what in #359 would produce such a major (5x) slow-down.

_Originally posted by @anthrotype in https://github.com/googlefonts/fontc/issues/365#issuecomment-1632981997_

**Comment by rsheeter** [comment]
+1 to worth it's own issue.

You can assign to me, slowdown from 359 is expected, I didn't have time to
tune prior to OOO and wanted to land as I'm quite confident in the general
direction of the refactoring. I also have some ideas about improving
visibility into how the work graph executed.

I would anticipate delay from 1) deliberately naive implementation (beware
of premature optimization and all that) and 2) changes to how glyph
processing threads: be glyphs current block in all fe glyphs, not just the
ones they depend on. Both issues should be fixable w/o inordinate issue;
the actual work to be done hasn't changed.

**Comment by rsheeter** [comment]
https://github.com/bheisler/iai might be worth a look; if it works we could have CI for perf for a small set of exemplar fonts.

**Comment by cmyr** [comment]
agree that this looks promising. I wonder if someone else has done the work to integrate that into github actions...

**Comment by rsheeter** [comment]
After #380 if I flamegraph Oswald (cmd below) the dominant factors are Font::load (16%) and and iup_delta_optimize (22%). It also looks like perhaps we would benefit from some buffer preallocation in glyph ir, will file a separate issue for that.

As none of this is new due to #359 I'd be inclined to call this fixed if the results hold up for Google Sans.

```shell
rm -rf build/ && cargo flamegraph -p fontc --root --  --source ../OswaldFont/sources/Oswald.glyphs --emit-ir false
```
